### ISSUE TITLE
Question: Defined Item in unidict recipe

### ISSUE THREAD

**Issue by Gandalf360one** [WanionCane/UniDict#190]
Hi I have a question.

Is it possible to have a defined item in a unidicted recipe ?
I tried something like this but it crashes my game:
mods.unidict.api.newShapedRecipeTemplate("dust", 1, [["ore", <immersiveengineering:tool:0>, null], [null, null, null], [null, null, null]]);

Sorry for the Bug tag.

**Comment by ElektroKill** [comment]
Hello, the craft tweaker code you posted is invalid because `"ore"` is not a valid recipe input. Also, I am pretty sure that `immersiveengineering:tool:0` should be encased by `<>` symbol creating `<immersiveengineering:tool:0>`.

By saying "defined item" do you mean a variable holding an ItemStack?
Providing a crash report would be helpful too ;) I will look into it as soon as possible.

**Comment by ElektroKill** [comment]
Hello, could you provide a crash report? I will try to look into this and if what you are trying to do is not supported I will be happy to implement it :)

**Comment by Gandalf360one** [comment]
Thanks for the reply.

I tried to apply the immersive engineering recipe (crushing an ore with the hammer to get 1 dust) to make it universal so that all ores will be effected that have a dust.
But my game crashed so I tried a bit around and it didn't work.
I look more into the recipes and recognized that immersive engineering already added all ores to the recipe so I realized that I don't have to add that recipe.

Long story short:
I tried to define the group ore and the hammer for the universal dust recipe but it didn't worked and I also don't need it (my fail)

Here is the crash.log if you want to look into it (edited the format from my script to make it right):
https://pastebin.com/uT5pyKq3

**Comment by ElektroKill** [comment]
Hello, there was actually a bug here, the detection of valid resources was flawed so it attempted to find ore variants of resources that don't have them, for example, it tried to find an Ore block for Lumium which doesn't exist causing a crash.

**Comment by ElektroKill** [comment]
I have fixed the bug in the upcoming 3.0.6 version
